apps/language_server/lib/language_server/dialyzer.ex

```diff
@@ -322,11 +344,15 @@ defmodule ElixirLS.LanguageServer.Dialyzer do
 
     {us, {active_plt, mod_deps, md5, warnings}} =
       :timer.tc(fn ->
+        Process.flag(:trap_exit, true)
+
         Task.async_stream(file_changes, fn {file, {content, _}} ->
           write_temp_file(root_path, file, content)
         end)
         |> Stream.run()
 
+        Process.flag(:trap_exit, false)
+
```

Comment:
this really should be a function 

```elixir
defp with_trapping_exits(fun) do 
  Process.flag(:trap_exit, true) 
  fun.() 
  Process.flag(:trap_exit, false) 
end
```

Reply:
I'd move `Process.flag(:trap_exit, false)` to `after` block